vllm/platforms/rocm.py

```diff
@@ -100,22 +101,33 @@ def wrapper(*args, **kwargs):
 
 @cache
 def on_gfx1x() -> bool:
-    GPU_ARCH = torch.cuda.get_device_properties("cuda").gcnArchName
+    GPU_ARCH = get_gcn_arch()
     return any(arch in GPU_ARCH for arch in ["gfx11", "gfx12"])
 
 
 @cache
 def on_mi3xx() -> bool:
-    GPU_ARCH = torch.cuda.get_device_properties("cuda").gcnArchName
+    GPU_ARCH = get_gcn_arch()
     return any(arch in GPU_ARCH for arch in ["gfx942", "gfx950"])
 
 
 @cache
 def on_gfx9() -> bool:
-    GPU_ARCH = torch.cuda.get_device_properties("cuda").gcnArchName
+    GPU_ARCH = get_gcn_arch()
     return any(arch in GPU_ARCH for arch in ["gfx90a", "gfx942", "gfx950"])
 
 
+def get_gcn_arch():
+    # Try to avoid torch cuda calls to avoid initializing CUDA
+    output = subprocess.check_output(['rocminfo']).decode()
```

Comment:
why not just rely on torch.cuda.get_device_properties("cuda").gcnArchName?

Also we can use @functools.lru_cache to cache the results.

Reply:
It's mainly because the vLLM needs to manipulate CUDA_VISIBLE_DEVICES after the process starts, but the manipulation will only be successful before CUDA/ROCm is initialized.

Some related issue / PR:
- https://github.com/pytorch/pytorch/issues/141678
- https://github.com/volcengine/verl/pull/1465
- https://github.com/vllm-project/vllm/issues/6056

About the fallback call, this is unfortunate because we want the check to be successful in the first place (since after all, users can set `RAY_EXPERIMENTAL_NOSET_ROCR_VISIBLE_DEVICES` to workaround this CUDA/ROCm initialization problem, but if you have a better option, please let me know!